pkg/controllers/provisioning/scheduling/volumetopology.go

```diff
@@ -153,44 +137,31 @@ func (v *VolumeTopology) getPersistentVolumeRequirements(ctx context.Context, po
 	return requirements, nil
 }
 
-func (v *VolumeTopology) getPersistentVolumeClaim(ctx context.Context, pod *v1.Pod, volume v1.Volume) (*v1.PersistentVolumeClaim, error) {
-	if volume.PersistentVolumeClaim == nil {
-		return nil, nil
-	}
-	pvc := &v1.PersistentVolumeClaim{}
-	if err := v.kubeClient.Get(ctx, types.NamespacedName{Name: volume.PersistentVolumeClaim.ClaimName, Namespace: pod.Namespace}, pvc); err != nil {
-		return nil, fmt.Errorf("getting persistent volume claim %q, %w", volume.PersistentVolumeClaim.ClaimName, err)
-	}
-	return pvc, nil
-}
-
 // ValidatePersistentVolumeClaims returns an error if the pod doesn't appear to be valid with respect to
 // PVCs (e.g. the PVC is not found or references an unknown storage class).
 func (v *VolumeTopology) ValidatePersistentVolumeClaims(ctx context.Context, pod *v1.Pod) error {
 	for _, volume := range pod.Spec.Volumes {
-		var storageClassName *string
-		var volumeName string
-		if volume.PersistentVolumeClaim != nil {
-			// validate the PVC if it exists
-			pvc, err := v.getPersistentVolumeClaim(ctx, pod, volume)
-			if err != nil {
+		pvc, err := volumeutil.GetPersistentVolumeClaim(ctx, v.kubeClient, pod, volume)
+		if err != nil {
+			return err
+		}
+		// may not have a pvc
+		if pvc == nil {
+			continue
+		}
+
+		if pvc.Spec.VolumeName != "" {
+			if err := v.validateVolume(ctx, pvc.Spec.VolumeName); err != nil {
 				return err
 			}
-			// may not have a PVC
-			if pvc == nil {
-				continue
-			}
-
-			storageClassName = pvc.Spec.StorageClassName
-			volumeName = pvc.Spec.VolumeName
-		} else if volume.Ephemeral != nil {
-			storageClassName = volume.Ephemeral.VolumeClaimTemplate.Spec.StorageClassName
+			continue
 		}
 
-		if err := v.validateStorageClass(ctx, storageClassName); err != nil {
-			return err
+		// PVC is unbound, we can't schedule unless the pod defines a valid storage class
+		if lo.FromPtr(pvc.Spec.StorageClassName) == "" {
```

Comment:
nit: store the storage class name in a variable and convert from a pointer one time?